skbeam/core/roi.py

```diff
@@ -430,18 +476,27 @@ def circular_average(image, calibrated_center, threshold=0, nx=100,
     ring_averages : array
         Radial average of the image. shape is (nx, ).
     """
-    radial_val = utils.radial_grid(calibrated_center, image.shape, pixel_size)
-
-    bin_edges, sums, counts = utils.bin_1D(np.ravel(radial_val),
-                                           np.ravel(image), nx,
+    radial_val = utils.radial_grid(calibrated_center, image.shape, pixel_size) 
+    if mask is not None:  
+        mask = np.array( mask, dtype = bool)
+        binr = radial_val[mask]
+        image_mask =     np.array( image )[mask]        
+    else:        
+        binr = np.ravel( radial_val ) 
+        image_mask = np.ravel(image) 
+    binr_ =   binr /(np.sqrt(pixel_size[1]*pixel_size[0] ))
+    bin_edges, sums, counts = bin_1D(      binr_,
+                                           image_mask,
+                                           nx=nx,
                                            min_x=min_x,
-                                           max_x=max_x)
+                                           max_x=max_x) 
     th_mask = counts > threshold
     ring_averages = sums[th_mask] / counts[th_mask]
-
     bin_centers = utils.bin_edges_to_centers(bin_edges)[th_mask]
+    return bin_centers, ring_averages 
```

Comment:
But counts is the counts of pixel per radius (so it's after the fact). I think it could potentially still be useful. Depending on your binning, you may want to ignore values that didn't average over enough pixels. I don't know if there is a point to returning the mask from that, just compare  threshold=0 with nonzero maybe?